src/main/java/org/fenixedu/academic/domain/student/Registration.java

```diff
@@ -3370,4 +3370,8 @@ public boolean isValidForRAIDES() {
         return FenixEduAcademicConfiguration.getConfiguration().getRaidesRequestInfo() && isActive() && isBolonha()
                 && !getDegreeType().isEmpty() && getRegistrationProtocol().isForOfficialMobilityReporting();
     }
+    
+    public ProgramConclusion getLastCurricularPlanProgramConclusion() {
+    	return getLastDegreeCurricularPlan().getLastOrderedCycleCourseGroup().getProgramConclusion();
+    }
```

Comment:
Please don't add more methods to this class. Move its implementation to where it is used.